### PR TITLE
Ltl recipe

### PR DESCRIPTION
**Pull request by teake**
@conda-forge/help-c-cpp

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ltl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - {{ name }}
+  ignore_run_exports:
+    # Both LAPACK and BLAS are needed during compilation for the test suite,
+    # but are not actually linked against in the LTL library (they are only
+    # needed in header files which are not compiled).
+    # If a downstream package actually needs the LTL marray/blas.h or
+    # marray/lapack.h headers, they should add libblas and / or liblapack to
+    # their requirements/host section.
+    - liblapack
+    - libblas
```
Comment (teake) [inline recipes/ltl/meta.yaml]: I'm not a 100% sure this is the correct way to go (I'm no C/C++/shared lib expert), but it seemed like a good solution.

File: recipes/ltl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  patches:
+    - 0001_Build_shared_lib.patch
+    - 0002_Fix_test_typo.patch
+    - 0003_Fix_blas_retun_type.patch
+
+build:
+  number: 0
+  skip: true  # [win]
+  run_exports:
+    - {{ name }}
```
Comment (chrisburr) [inline recipes/ltl/meta.yaml]: Should this constrain the version for the export, with something like `- {{ pin_subpackage(name, max_pin='x.x') }}`? Where the number of `x`s indicates the ABI stability

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements

Comment (teake) [reply recipes/ltl/meta.yaml]: Good idea, I missed that. I just pushed a fixing commit.

**Comment by scopatz** [comment]
This looks good to me!

**Pull request merged**